src/components/button/button-styles.jsx

```diff
@@ -106,27 +106,88 @@ export default theme => {
       fontWeight: theme.typography.fontWeightLight,
       borderRadius: 12,
       padding: '5px 12px',
+
+      '&.iconText': {
+        paddingBottom: 5,
+        paddingTop: 4,
+      },
+      '&.icon': {
+        padding: 4,
+      },
+      '& svg': {
+        height: 11,
+        width: 11,
+      },
     },
 
     sm: {
       fontSize: 11,
       fontWeight: theme.typography.fontWeightLight,
       borderRadius: 16,
       padding: '8px 20px',
+
+      '&.iconText': {
+        paddingBottom: 7,
+        paddingTop: 7,
+      },
+      '&.icon': {
+        padding: 6,
+      },
+      '& svg': {
+        height: 13,
+        width: 13,
+      },
     },
 
     md: {
       fontSize: 14,
       fontWeight: theme.typography.fontWeightRegular,
       borderRadius: 20,
       padding: '11px 38px',
+
+      '&.iconText': {
+        paddingBottom: 10,
+        paddingTop: 10,
+      },
+      '&.icon': {
+        padding: 8,
+      },
+      '& svg': {
+        height: 16,
+        width: 16,
+      },
     },
 
     lg: {
       fontSize: 16,
       fontWeight: theme.typography.fontWeightRegular,
       borderRadius: 24,
       padding: '14px 40px',
+
+      '&.iconText': {
+        paddingBottom: 13,
+        paddingTop: 13,
+      },
+      '&.icon': {
+        padding: 10,
+      },
+      '& svg': {
+        height: 18,
+        width: 18,
+      },
+    },
+
+    '.icon': {
```

Comment:
You can use JSS for these *(icon, iconText)* also. To refer to them inside other elements (such as `'&.icon'`) you simply use `'&$icon'` instead :)

Reply:
Oooh, I didn't know about `'&$icon'`. I originally had them under `classes` and was searching around for how to make it work, but settled on following the current pattern. I'll refactor to that!

Reply:
The only issue is that `iconText` has no default styles, only styles for each size. I could implement it with an empty prop:
```
iconText: {},
```
Seems a bit hacky, but it'd work.